experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -127,6 +140,9 @@ func (r *ExpanderReconciler) getPlanForInputCR(ctx context.Context, inputcr *uns
 				"name", plancr.Name, "namespace", plancr.Namespace)
 			return planNN, &plancr, err
 		}
+		// Add a finalizer to the Plan to discourage out of band deletions. Stage information is
+		// stored in the Plan to handle proper cleanup of resources when a Facade is deleted.
+		controllerutil.AddFinalizer(&plancr, finalizerName)
```

Comment:
Would we need this in the else case as well ? 
Existing Plan CRs ? 
I dont think it is a required. Just a thought.

Reply:
Yeah, it can't hurt to check pre-existing plans and ensure they have the finalizer as well.